applications/Chat/coati/models/lora.py

```diff
@@ -61,7 +61,12 @@ def T(w):
         if self.merge_weights and self.merged:
             # Make sure that the weights are not merged
             if self.r > 0:
-                self.weight.data -= T(self.lora_B @ self.lora_A) * self.scaling
+                if not hasattr(self, "lora_A") or not hasattr(self, "lora_B"):
+                    self.lora_A = nn.Parameter(self.weight.new_zeros((self.r, self.in_features)))
```

Comment:
When doesn't it  have lora_A or lora_B when r>0?

Reply:
they are deleted in eval() when self.merge_weights=True

Reply:
Note that this branch is run when weight is merged. Why we need a new lora_A?

Reply:
If you want to re-train after inference, you should never merge weight

Reply:
I need to send the updated weights to remote experience makers

Reply:
You can send lora weight alone without merging

Reply:
Besides, a new lora weight is different from the old one. This may affect convergence.

Reply:
I suggest leaving lora integration to the next PR. We quickly support naive fine-tuning first.